### PR TITLE
mon/MonClient: send logs to mon on separate schedule than pings

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
as mentioned the other day over IRC.

File: src/mon/MonClient.cc [hunk 1]
```diff
@@ -874,53 +889,45 @@ void MonClient::_finish_hunting(int auth_err)
...
+  global_id = active_con->get_global_id();
+  do_send_ping = make_lambda_context(
+    [this](int) {
+      send_ping();
+      _maybe_renew_subs();
+      _check_auth_tickets();
+    });
+  timer.add_event_after(cct->_conf->mon_client_ping_interval, do_send_ping);
+  do_send_log = make_lambda_context([this](int) { send_log(); });
+  timer.add_event_after(cct->_conf->mon_client_log_interval, do_send_log);
```
Comment (liewegas) [inline src/mon/MonClient.cc]: I'm not seeing where another event is scheduled when do_send_ping or do_send_log runs?

Comment (tchaikov) [reply src/mon/MonClient.cc]: updated and repushed.

**Comment by tchaikov** [comment]
changelog

- drop the change adding `src/test/cli/osdmaptool/Testing/Temporary/CTestCostData.txt`

**Comment by liewegas** [comment]
```
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:     -1> 2020-03-11T04:18:44.653+0000 7feacffb7700 -1 /build/ceph-15.1.0-1993-g94ee50c/src/mon/MonClient.cc: In function 'void MonClient::_maybe_renew_subs()' thread 7feacffb7700 time 2020-03-11T04:18:44.638065+0000
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc: /build/ceph-15.1.0-1993-g94ee50c/src/mon/MonClient.cc: 957: FAILED ceph_assert(active_con)
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  ceph version 15.1.0-1993-g94ee50c (94ee50cc1178862d969242b021f09c45996feb22) octopus (rc)
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  1: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x154) [0x7fec2bd4d0b0]
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  2: (ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char const*, ...)+0) [0x7fec2bd4d28b]
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  3: (MonClient::_maybe_renew_subs()+0x1f6) [0x7fec2c090506]
2020-03-11T04:18:44.716 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  4: (MonClient::tick()+0x11) [0x7fec2c090541]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  5: (MonClient::_finish_hunting(int)+0x475) [0x7fec2c092705]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  6: (MonClient::handle_auth(MAuthReply*)+0x34a) [0x7fec2c09738a]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  7: (MonClient::ms_dispatch(Message*)+0x27b) [0x7fec2c09773b]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  8: (DispatchQueue::entry()+0xdfa) [0x7fec2bf4155a]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  9: (DispatchQueue::DispatchThread::entry()+0xd) [0x7fec2bfe10ed]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  10: (()+0x76db) [0x7fec349546db]
2020-03-11T04:18:44.717 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  11: (clone()+0x3f) [0x7fec2ae6788f]
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:      0> 2020-03-11T04:18:44.677+0000 7feacffb7700 -1 *** Caught signal (Aborted) **
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  in thread 7feacffb7700 thread_name:ms_dispatch
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  ceph version 15.1.0-1993-g94ee50c (94ee50cc1178862d969242b021f09c45996feb22) octopus (rc)
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  1: (()+0x12890) [0x7fec3495f890]
2020-03-11T04:18:44.718 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  2: (gsignal()+0xc7) [0x7fec2ad84e97]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  3: (abort()+0x141) [0x7fec2ad86801]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  4: (ceph::__ceph_assert_fail(char const*, char const*, int, char const*)+0x1a5) [0x7fec2bd4d101]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  5: (ceph::__ceph_assertf_fail(char const*, char const*, int, char const*, char const*, ...)+0) [0x7fec2bd4d28b]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  6: (MonClient::_maybe_renew_subs()+0x1f6) [0x7fec2c090506]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  7: (MonClient::tick()+0x11) [0x7fec2c090541]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  8: (MonClient::_finish_hunting(int)+0x475) [0x7fec2c092705]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  9: (MonClient::handle_auth(MAuthReply*)+0x34a) [0x7fec2c09738a]
2020-03-11T04:18:44.719 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  10: (MonClient::ms_dispatch(Message*)+0x27b) [0x7fec2c09773b]
2020-03-11T04:18:44.720 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  11: (DispatchQueue::entry()+0xdfa) [0x7fec2bf4155a]
2020-03-11T04:18:44.720 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  12: (DispatchQueue::DispatchThread::entry()+0xd) [0x7fec2bfe10ed]
2020-03-11T04:18:44.720 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  13: (()+0x76db) [0x7fec349546db]
2020-03-11T04:18:44.720 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  14: (clone()+0x3f) [0x7fec2ae6788f]
2020-03-11T04:18:44.720 INFO:tasks.workunit.client.0.smithi012.stdout:                 api_misc:  NOTE: a copy of the executable, or `objdump -rdS <executable>` is needed to interpret this.
```
/a/sage-2020-03-11_01:39:50-rados-wip-sage-testing-2020-03-10-1722-distro-basic-smithi/4844783

**Comment by liewegas** [comment]
i'm going to merge the other one for now... we can improve it with this later.

**Comment by smithfarm** [comment]
@tchaikov Ping?

**Comment by tchaikov** [comment]
@smithfarm thanks for the reminder. will update this PR very soon.

**Comment by smithfarm** [comment]
@tchaikov Is it useful to keep this PR open when it is targeting the wrong branch? There is a (small) danger it might get merged by mistake.

When I run `ceph-backport.sh --milestones` to check the milestone setting on backport PRs, this PR always gets flagged.

**Pull request closed without merging**